### PR TITLE
[INTENG-4322] feat: added initSessionWithoutState

### PR DESCRIPTION
**Pull request by code-crusher**
This is to support the cases where you finish the Activity just after initSession. Common case: Splash launcher without any UI.

### CODE REVIEW

**Comment by aaaronlopez** [comment]
Why not just have a setter for `bypassCurrentActivityIntentState_` and add a check for it in this section? https://github.com/BranchMetrics/android-branch-deep-linking/blob/3a222245fa061c3673c4fd7a1f354f4451dda467/Branch-SDK/src/io/branch/referral/Branch.java#L441-L448

**Comment by code-crusher** [comment]
@aaaronlopez sounds good! We can do it. Any chance we also do QA for this? Wanna make sure we are not leaving any edge case for OLA.

**Comment by aaaronlopez** [comment]
Let's talk to @yury-branch and @GeneShay

**Comment by Sarkar** [comment]
@apeterson-branch -- if you could take a quick peek at this as well.

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -1061,6 +1063,25 @@ public boolean initSessionForced(BranchReferralInitListener callback) {
...
+     * <p>Force initialises a session with the Branch API with bypassing the intentState_ READY callback from the current Activity, assigning a {@link BranchReferralInitListener}
+     * to perform an action upon successful initialisation.</p>
+     *
+     * @param callback A {@link BranchReferralInitListener} instance that will be called following
+     *                 successful (or unsuccessful) initialisation of the session with the Branch API.
+     * @return A {@link Boolean} value, indicating <i>false</i> if initialisation is
+     * unsuccessful.
+     */
+
+    public boolean initSessionWithoutState(BranchReferralInitListener callback) {
```
Comment (aaaronlopez) [inline Branch-SDK/src/io/branch/referral/Branch.java]: @code-crusher can we find a way to get rid of this and just use the boolean `bypassCurrentActivityIntentState_`?

**Comment by aaaronlopez** [comment]
looks great! 👍 let's not merge this just yet

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 2]
```diff
@@ -3174,6 +3185,10 @@ public static void disableForcedSession() {
     public static boolean isForceSessionEnabled() {
         return isForcedSession_;
     }
+    
+    public static void BypassCurrentActivityIntentState_() {
```
Comment (aaaronlopez) [inline Branch-SDK/src/io/branch/referral/Branch.java]: can we rename this `enableBypassCurrentActivityIntentState`?

**Pull request merged**